### PR TITLE
Electricalusage

### PR DESCRIPTION
**Pull request by miguelut**
For code review.

### CODE REVIEW

File: lambdas/electricalusage/genability/hello.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (miguelut) [inline lambdas/electricalusage/genability/hello.py]: We can probably remove this file.

**Comment by miguelut** [comment]
We should genability related code to its own repo and then have a build script that pulls it down into a directory that is .gitignored.  The genability folder and the genability_api.py file will be shared between several lambdas.  Unfortunately lambda doesn't make it easy for us to share code, so we need to sort of take care of that with a build/package script.

**Comment by miguelut** [comment]
Need to add .DS_Store to the .gitignore

File: lambdas/electricalusage/index.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import json
+import datetime
+import genability_api
+from genability.account import Account
+from genability import profile
+
+def handler(event, context):
+    create_response = lambda status_code, body : { 'statusCode': status_code, 'body': json.dumps(body)}
```
Comment (miguelut) [inline lambdas/electricalusage/index.py]: Don't use a lambda proxy here.  Instead, configure API Gateway with a lambda integration and map your models/exceptions to HTTP status codes.  This decouples our webserver logic from our API logic.  You can configure this stuff from the AWS UI or I believe you can also configure them in the template.yml file.

File: template.yml [hunk 3]
```diff
@@ -11,30 +10,26 @@ Parameters:
...
-#   Function:
-#     AutoPublishAlias: live
-#     DeploymentPreference:
-#       Enabled: true
-#       Type: Canary10Percent5Minutes
+Globals:
+  Function:
+    Environment:
+      Variables:
+        GENABILITY_APP_ID: 5bb07e76-acb5-4ca0-b79b-d53cbc498873
```
Comment (miguelut) [inline template.yml]: Do we want this info in the repo?  Is it sensitive data?

**Comment by miguelut** [comment]
In general I'd say we probably want to build out the SAM template/swagger definition a bit more, even though a lot of it will be repeating what's in the Genability API for this specific call.

File: lambdas/electricalusage/genability/hello.py (hunk 1 above)
Comment (xCmac) [reply lambdas/electricalusage/genability/hello.py]: Agreed, it was just copied over together from QuotingService project.

File: lambdas/electricalusage/index.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        )
+        try:
+            usage = genability_api.update_usage(usage_profile)
+            return create_response(200, usage)
+        except GenabilityError as e:
+            logging.error(e.metadata)
+            return create_response(500, e.metadata)
+    else:
+        logging.error('Error 500: Invalid Arguments')
+        return create_response(500, 'Error 500: Invalid Arguments')
```
Comment (miguelut) [inline lambdas/electricalusage/index.py]: So I know I said make everything a 500, but this is exactly a 400 (i.e. the data the frontend sent us is bad.)

**Comment by xCmac** [comment]
@miguelut using this same branch to add solar production profile, so don't merge yet.

**Comment by miguelut** [comment]
Closing this out because these changes are encompassed in the other pull request.

**Pull request closed without merging**